src/metrics/nom.rs

```diff
@@ -97,12 +108,41 @@ impl Nom for MozjsCode {
         use Mozjs::*;
 
         match node.object().kind_id().into() {
-            Function | FunctionDeclaration | MethodDefinition => {
+            FunctionDeclaration | MethodDefinition => {
                 stats.functions += 1;
             }
-            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+            GeneratorFunction | GeneratorFunctionDeclaration => {
                 stats.closures += 1;
             }
+            Function => {
+                // Consider named functions as functions and unnamed ones as
+                // closures.
+                if count_specific_ancestors!(
+                    node,
+                    VariableDeclarator | AssignmentExpression | LabeledStatement | Pair,
+                    StatementBlock | ReturnStatement | NewExpression | Arguments
+                ) > 0
+                    || is_child_identifier(node, Identifier as u16)
+                {
+                    stats.functions += 1;
+                } else {
+                    stats.closures += 1;
+                }
+            }
+            ArrowFunction => {
```

Comment:
What if you handled ArrowFunction in exactly the same way as Function?

Reply:
`javascript_arrow_new_nom` and `javascript_arrow_nom` fail

Reply:
OK, not feasible then.
Could you move the `count_specific_ancestors!(` part in a helper function, so we don't have to duplicate it?